language-server/src/server.js

```diff
@@ -107,6 +116,30 @@ connection.onInitialized(async () => {
   await validateWorkspace();
 });
 
+const documentSettings = new Map(); // Consider a Map for cache
+let globalSettings = {
+  enableDetailedErrors: false,
+  schemaValidationSeverity: "error",
+  defaultDialect: "http://json-schema.org/draft-07/schema#"
```

Comment:
There shouldn't be a default value for this config. A default dialect should only apply if the user explicitly sets a default in their config.

```suggestion
```